ocaml/typing/jkind.ml

```diff
@@ -1443,46 +1755,19 @@ let report_error ~loc = function
         (* CR layouts errors: use the context to produce a better error message.
            When RAE tried this, some types got printed like [t/2], but the
            [/2] shouldn't be there. Investigate and fix. *)
-        "@[<v>Layout %s is more experimental than allowed by the enabled \
+        "@[<v>Layout %a is more experimental than allowed by the enabled \
          layouts extension.@;\
          %t@]"
-        (Legacy.string_of_const jkind)
-        hint)
+        Const.format jkind hint)
 
 let () =
   Location.register_error_of_exn (function
-    | User_error (loc, err) -> Some (report_error ~loc err)
+    | Error.User_error (loc, err) -> Some (report_error ~loc err)
     | _ -> None)
 
 (* CR layouts v2.8: Remove the definitions below by propagating changes
    outside of this file. *)
 
-type const = Legacy.const =
-  | Any
-  | Value
-  | Void
-  | Immediate64
-  | Immediate
-  | Float64
-  | Float32
-  | Word
-  | Bits32
-  | Bits64
-
-type annotation = const * Jane_syntax.Jkind.annotation
-
-let string_of_const = Legacy.string_of_const
-
-let equal_const = Legacy.equal_const
-
-type desc =
-  | Const of const
-  | Var of Sort.var
-
-let to_legacy_desc : Desc.t -> desc = function
-  | Const c -> Const (Const.to_legacy_jkind c)
-  | Var v -> Var v
-
-let get t = to_legacy_desc (get t)
+type annotation = Const.t * Jane_syntax.Jkind.annotation
```

Comment:
I think it should be a goal of this work to get rid of this. That is, once we have a decent printing story, we shouldn't have to remember the user-written bit.

Reply:
This should be removed when printing is improved in the near future. I'll add a CR.